packages/aws-cdk/test/api/logs/register-cloudwatch-logs.test.ts

```diff
@@ -0,0 +1,241 @@
+  // GIVEN
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        LogGroup: {
+          Properties: {
+            Type: 'AWS::Logs::LogGroup',
+            LogGroupName: 'log_group',
+          },
+        },
+        Def: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            LoggingConfiguration: {
+              Destinations: [
+                {
+                  CloudWatchLogsLogGroup: {
+                    LogGroupArn: {
+                      'Fn::GetAtt': ['LogGroup', 'Arn'],
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+    },
+  });
+  pushStackResourceSummaries(stackSummaryOf('Def', 'AWS::StepFunctions::StateMachine', 'machine'));
+  pushStackResourceSummaries(stackSummaryOf('LogGroup', 'AWS::Logs::LogGroup', 'log_group'));
+
+  // WHEN
+  const result = await registerCloudWatchLogGroups(logsMockSdkProvider.mockSdkProvider, cdkStackArtifact);
+
+  // THEN
+  expect(result.logGroupNames).toEqual(['log_group']);
+});
+
+test('excluded log groups are not added', async () => {
+  // GIVEN
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        LogGroup: {
+          Properties: {
+            Type: 'AWS::Logs::LogGroup',
+            LogGroupName: 'log_group',
+          },
+        },
+        LogGroup2: {
+          Properties: {
+            Type: 'AWS::Logs::LogGroup',
+            LogGroupName: 'log_group2',
+          },
+        },
+        Def: {
+          Type: 'AWS::CodeBuild::Project',
+          Properties: {
+            PojectName: 'project',
+          },
+        },
+        FlowLog: {
+          Type: 'AWS::EC2::FlowLog',
+          Properties: {
+            LogDestination: { Ref: 'LogGroup' },
+          },
+        },
+        FlowLog2: {
+          Type: 'AWS::EC2::FlowLog',
+          Properties: {
+            LogDestination: {
+              'Fn::GetAtt': ['LogGroup2', 'Arn'],
+            },
+          },
+        },
+      },
+    },
+  });
+  pushStackResourceSummaries(stackSummaryOf('FlowLog', 'AWS::EC2::FlowLog', 'flow'));
+  pushStackResourceSummaries(stackSummaryOf('Def', 'AWS::CodeBuild::Project', 'project'));
+  pushStackResourceSummaries(stackSummaryOf('LogGroup', 'AWS::Logs::LogGroup', 'log_group'));
+  pushStackResourceSummaries(stackSummaryOf('LogGroup2', 'AWS::Logs::LogGroup', 'log_group2'));
+
+  // WHEN
+  const result = await registerCloudWatchLogGroups(logsMockSdkProvider.mockSdkProvider, cdkStackArtifact);
+
+  // THEN
+  expect(result.logGroupNames).toEqual([]);
+});
+
+test('unassociated log groups are added', async () => {
+  // GIVEN
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        LogGroup: {
+          Properties: {
+            Type: 'AWS::Logs::LogGroup',
+            LogGroupName: 'log_group',
```

Comment:
There should probably be at lest one test when the `LogGroupName` is _not_ provided.